src/ucp/core/ucp_worker.c

```diff
@@ -1209,21 +1209,28 @@ static void ucp_worker_iface_set_sys_device_distance(ucp_worker_iface_t *wiface)
     ucs_sys_dev_distance_t *distance = &wiface->distance;
     ucs_sys_device_t device          = UCS_SYS_DEVICE_ID_UNKNOWN;
     ucs_sys_device_t cmp_device      = UCS_SYS_DEVICE_ID_UNKNOWN;
-    ucp_rsc_index_t md_index, i;
+    ucp_rsc_index_t i;
+    ucp_tl_resource_desc_t *cmp_rsc;
+    char buf[128];
 
     *distance = ucs_topo_default_distance;
 
     for (i = 0; i < context->num_tls; i++) {
-        md_index = context->tl_rscs[i].md_index;
-        if (strcmp(context->tl_mds[md_index].rsc.md_name,
+        cmp_rsc = &context->tl_rscs[i];
+        if (strcmp(context->tl_mds[cmp_rsc->md_index].rsc.md_name,
                    context->config.ext.select_distance_md)) {
             continue;
         }
 
         device     = ucp_worker_iface_get_sys_device(wiface);
-        cmp_device = context->tl_rscs[i].tl_rsc.sys_device;
+        cmp_device = cmp_rsc->tl_rsc.sys_device;
 
         ucs_topo_get_distance(device, cmp_device, distance);
+        ucs_trace("distance between %s/dev[%d] and %s/dev[%d] is %s",
+                  context->tl_rscs[wiface->rsc_index].tl_rsc.tl_name, device,
```

Comment:
WDYT about using UCT_TL_RESOURCE_FMT?

Reply:
If we use `UCT_TL_RESOURCE_DESC_FMT`, then CUDA devices will not contain device index. E.g.:
```
ucp_worker.c:1234 UCX  TRACE distance between rc_mlx5/mlx5_0:1 and cuda_copy/cuda is 500ns 5100.00MB/s
```

Reply:
WDYT think about using both `UCT_TL_RESOURCE_FMT` and sys_device? E.g.
```
ucs_trace("distance between "UCT_TL_RESOURCE_FMT"/dev[%d] and "UCT_TL_RESOURCE_FMT"/dev[%d] is %s",...)
```

Other idea is extend the `UCT_TL_RESOURCE_FMT` so it will print `sys_device` if it's not unknown. Or introduce the new macro that will do it.

Reply:
Updated. grep log by "distance between" may look like this:
```
ucp_worker.c:1249 UCX  TRACE distance between self/memory and cuda_copy/cuda/dev[0] is 0ns >1PB/s
ucp_worker.c:1249 UCX  TRACE distance between tcp/ens10f0/dev[1] and cuda_copy/cuda/dev[0] is 500ns 5100.00MB/s
ucp_worker.c:1249 UCX  TRACE distance between tcp/lo and cuda_copy/cuda/dev[0] is 0ns >1PB/s
ucp_worker.c:1249 UCX  TRACE distance between tcp/ib1/dev[2] and cuda_copy/cuda/dev[0] is 500ns 5100.00MB/s
ucp_worker.c:1249 UCX  TRACE distance between sysv/memory and cuda_copy/cuda/dev[0] is 0ns >1PB/s
ucp_worker.c:1249 UCX  TRACE distance between posix/memory and cuda_copy/cuda/dev[0] is 0ns >1PB/s
ucp_worker.c:1249 UCX  TRACE distance between cuda_copy/cuda/dev[0] and cuda_copy/cuda/dev[0] is 0ns >1PB/s
ucp_worker.c:1249 UCX  TRACE distance between cuda_ipc/cuda/dev[0] and cuda_copy/cuda/dev[0] is 0ns >1PB/s
ucp_worker.c:1249 UCX  TRACE distance between gdr_copy/cuda/dev[0] and cuda_copy/cuda/dev[0] is 0ns >1PB/s
ucp_worker.c:1249 UCX  TRACE distance between dc_mlx5/mlx5_0:1/dev[3] and cuda_copy/cuda/dev[0] is 500ns 5100.00MB/s
ucp_worker.c:1249 UCX  TRACE distance between rc_verbs/mlx5_0:1/dev[3] and cuda_copy/cuda/dev[0] is 500ns 5100.00MB/s
ucp_worker.c:1249 UCX  TRACE distance between rc_mlx5/mlx5_0:1/dev[3] and cuda_copy/cuda/dev[0] is 500ns 5100.00MB/s
ucp_worker.c:1249 UCX  TRACE distance between ud_verbs/mlx5_0:1/dev[3] and cuda_copy/cuda/dev[0] is 500ns 5100.00MB/s
ucp_worker.c:1249 UCX  TRACE distance between ud_mlx5/mlx5_0:1/dev[3] and cuda_copy/cuda/dev[0] is 500ns 5100.00MB/s
ucp_worker.c:1249 UCX  TRACE distance between dc_mlx5/mlx5_1:1/dev[2] and cuda_copy/cuda/dev[0] is 500ns 5100.00MB/s
ucp_worker.c:1249 UCX  TRACE distance between rc_verbs/mlx5_1:1/dev[2] and cuda_copy/cuda/dev[0] is 500ns 5100.00MB/s
ucp_worker.c:1249 UCX  TRACE distance between rc_mlx5/mlx5_1:1/dev[2] and cuda_copy/cuda/dev[0] is 500ns 5100.00MB/s
ucp_worker.c:1249 UCX  TRACE distance between ud_verbs/mlx5_1:1/dev[2] and cuda_copy/cuda/dev[0] is 500ns 5100.00MB/s
ucp_worker.c:1249 UCX  TRACE distance between ud_mlx5/mlx5_1:1/dev[2] and cuda_copy/cuda/dev[0] is 500ns 5100.00MB/s
ucp_worker.c:1249 UCX  TRACE distance between cma/memory and cuda_copy/cuda/dev[0] is 0ns >1PB/s
ucp_worker.c:1249 UCX  TRACE distance between knem/memory and cuda_copy/cuda/dev[0] is 0ns >1PB/s
ucp_worker.c:1249 UCX  TRACE distance between xpmem/memory and cuda_copy/cuda/dev[0] is 0ns >1PB/s